qa/tasks/keycloak.py

```diff
@@ -0,0 +1,331 @@
+"""
+Deploy and configure Keycloak for Teuthology
+"""
+from io import BytesIO
+from io import StringIO
+from configobj import ConfigObj
+import base64
+import argparse
+import contextlib
+import logging
+import os
+import random
+import six
+import string
+import subprocess
+import json
+import sys
+from pathlib import Path
+
+from collections import OrderedDict
+from itertools import chain
+
+from teuthology import misc as teuthology
+from teuthology import contextutil
+from teuthology.config import config as teuth_config
+from teuthology.orchestra import run
+from teuthology.packaging import install_package
+from teuthology.packaging import remove_package
+from teuthology.exceptions import ConfigError
+
+log = logging.getLogger(__name__)
+
+def get_keycloak_dir(ctx):
+    return '{tdir}/keycloak-11.0.0'.format(tdir=teuthology.get_testdir(ctx))
```

Comment:
@alimaredia : I had a question here, not sure if Kalpesh has already asked this - the keycloak version is hardcoded here, should we look at checking for a way to get the latest stable version, or should we just leave this as is, and change the version here from time to time?

Reply:
Good catch, no he did not mention that to me. Could we pass in the key cloak version as a client config variable. I see "11.0.0" hard coded lots of places.

Reply:
Ahh sorry I forgot to discuss this with @alimaredia. I tried to check if we could do that way but for every version they keep tar.gz file and that only has a mention of this version detail like "11.0.0". So, I think we can update this as and when needed.

Reply:
I understand that we might keep it at 11.0.0 for a while, but we still shouldn't be hard coding it in anywhere. It should be passed in via a config variable so when the version does need to be updated, all you have to do is replace the .yaml file.

Reply:
Okay.

Reply:
I still see `11-0-0` hard coded in this PR. It should be passed into the .yaml file that's calling the `keycloak` task.